lib/driver.js

```diff
@@ -22,29 +24,43 @@ class AndroidDriver extends BaseDriver {
     [sessionId] = await super.createSession(caps);
 
     // TODO fail if app file is missing, or if app is not on the device?
-
+    // assigning defaults
+    let defaultOpts = {action: "android.intent.action.MAIN",
+                       category: "android.intent.category.LAUNCHER",
+                       flags: "0x10200000",
+                       disableAndroidWatchers: false,
+                       tmpDir: await tempDir.staticDir()};
+    _.defaults(this.opts, defaultOpts);
     if (!this.opts.javaVersion) {
       this.opts.javaVersion = await helpers.getJavaVersion();
     }
-
+    // Add condition for chrome driver
+    if (this.opts.app) {
+      // download and unzip
+      this.opts.app = await this.helpers.configureApp(this.opts.app, APP_EXTENSION);
+    }
+    await this.checkAppPresent();
     await this.startAndroidSession(this.opts);
-
     return [sessionId, caps];
   }
 
+  async onSettingsUpdate (update) {
```

Comment:
looking at this now, and since we're not using an event emitter, i'm wondering if this should work more like:

```
async onSettingsUpdate (key, value) {
```